lib/utils.c

```diff
@@ -39,6 +39,56 @@
 
 int timestamp_short;
 
+int read_prop(char *dev, char *prop, long *value)
+{
+	char fname[128], buf[80], *endp;
+	int status;
+	FILE *fp;
+	long result;
+	int ret;
+
+	ret = snprintf(fname, sizeof(fname), "/sys/class/net/%s/%s", dev, prop);
+	if (ret <= 0 || ret >= sizeof(fname))
+		goto out;
+
+	fp = fopen(fname, "r");
+	if (fp == NULL) {
+		fprintf(stderr, "fopen %s: %s\n", fname, strerror(errno));
+		return -1;
+	}
+
+	status = fread(buf, sizeof(buf), 1, fp);
+	if (status < 0) {
+		fprintf(stderr, "fread %s: %s", fname, strerror(errno));
+		return -1;
+	}
+
+	if (status != sizeof(buf)) {
+		fprintf(stderr, "Unexpected EOF during fread %s: %s", fname, strerror(errno));
+	}
+
+	fclose(fp);
+	result = strtoul(buf, &endp, 0);
```

Comment:
I don't think the handling below is exactly right. I think we just want:

    result = strtoul(buf, &endp, 0);
    if (buf == endp || *endp) {
    	fprintf("value \"%s\" in file %s is not a number\n", buf, fname);
    	goto out;
    }
    if (result == ULONG_MAX && errno == ERANGE) {
    	fprintf(stderr, "strtoul %s: %s", fname, strerror(errno));
    	goto out;
    }

Reply:
But shouldn't we do something when the nothing is interpreted by `strtoul()` , which happens when the file has an empty string (cases like wlan0 or bridge devices, etc)? The `result == 0` case takes care of that. Let me know what you think.

Reply:
But `fgets()` would catch that case now, so  I don't think we need `result == 0` case anymore, will make the change